### PR TITLE
 Feat: Best Practices Section

### PR DESCRIPTION
**Pull request by renatamottam**
## What's the purpose of this pull request?
Following [Fanny's suggestion](https://github.com/vtex-sites/nextjs.store/pull/94#discussion_r889175592), we decided to create a Best Practices section that will first be used at `Alert` doc page.

Also, I updated `Alert` doc page to use `TokenTable`and suggested a new `valueGlobal` prop for when we need to reference a color that's different from `value` prop.

## How does it work?
When needed, you can import `BestPractices` and `BestPracticesRule` from `'src/../.storybook/components'` and use it on your `stories.mdx`:
```
<BestPractices>
  <BestPracticesRule
    recommendedUsage={<RecommendedComponentExample />}
    recommendedDescription="..."
    discouragedUsage={<DiscouragedComponentExample />}
    discouragedDescription="..."
  />
</BestPractices>
```

The result should be something like this:

image

## How to test it?
You can see and usage example on [Alert doc page](https://nextjs-store-storybook-git-feat-best-practices-504117-faststore.vercel.app/?path=/docs/molecules-alert--default-story#dos-and-donts).

## References
https://style.monday.com/?path=/docs/feedback-alertbanner--overview

## Checklist

**Changelog**
- [x] Added an entry in the `CHANGELOG.md` at the beginning of its due section. [The latest version should comes first](https://keepachangelog.com/en/1.0.0/#:~:text=The%20latest%20version%20comes%20first.).
- [x] Added the PR number with the PR link at the entry in the `CHANGELOG.md`. E.g., *New items in the `pull_request_template.md` ([#4](https://github.com/vtex-sites/nextjs.store/pull/4))* 

**PR Description**
- [x] Added a label according to the PR goal - `Breaking change`, `Enhancement`, `Bug` or `Chore`.

**Documentation**
- [x] PR description
- [x] Added to/Updated the Storybook - *if applicable*.
- [x] For documentation changes, ping @ carolinamenezes, @ PedroAntunesCosta or @ Mariana-Caetano to review and update.

### CODE REVIEW

File: .storybook/components/BestPracticesRule.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import React, { ReactNode } from 'react'
+
+type BestPracticesRuleProps = {
+  componentGood: ReactNode
+  componentBad: ReactNode
```
Comment (hellofanny) [inline .storybook/components/BestPracticesRule.tsx:5]:
```suggestion
  discourageUsage: ReactNode
```

File: .storybook/components/TokenRow.tsx [hunk 2]
```diff
@@ -5,16 +5,24 @@ type TokenRowProps = {
...
   value: string
   isColor?: boolean
+  valueGlobal?: string
 }
 
-const TokenRow = ({ token, value, isColor = false }: TokenRowProps) => {
+const TokenRow = ({
+  token,
+  value,
+  valueGlobal,
```
Comment (hellofanny) [inline .storybook/components/TokenRow.tsx:14]:
```suggestion
  globalValue,
```

File: .storybook/components/BestPractices.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React, { ReactNode } from 'react'
+
+type BestPracticesProps = {
+  title?: string
+  description?: string
+  children: ReactNode
+}
+
+const BestPractices = ({ children }: BestPracticesProps) => {
```
Comment (eduardoformiga) [inline .storybook/components/BestPractices.tsx:9]: Hmm, what do you think about using `PropsWithChildren`?

```suggestion
import React from 'react'
import type { PropsWithChildren } from 'react'

type BestPracticesProps = {
  title?: string
  description?: string
}

const BestPractices = ({ children }: PropsWithChildren<BestPracticesProps>) => {
```

File: .storybook/components/TokenRow.tsx [hunk 4]
```diff
@@ -5,16 +5,24 @@ type TokenRowProps = {
...
+}: TokenRowProps) => {
   return (
     <tr>
       <td>
         <Code>{token}</Code>
       </td>
       <td>
-        {isColor && <div style={{ backgroundColor: value }} />}
+        {isColor && (
+          <div style={{ backgroundColor: valueGlobal ? valueGlobal : value }} />
```
Comment (eduardoformiga) [inline .storybook/components/TokenRow.tsx:24]: if you agree with @hellofanny's suggestion, make sure to update here as well :)

**Review by eduardoformiga — COMMENTED** [review]
Awesome! we should add a Renatinha full-stack stamp here hahah

Left some minor comments.

File: .storybook/components/TokenRow.tsx (hunk 2 above)
Comment (ArthurTriis1) [reply .storybook/components/TokenRow.tsx]: I think globalValue is just used in backgroundColor right? So can we call this prop "colorValue"? If yes, we can change the logic to show the color also. Like this: 

```jsx
{ ( isColor || colorValue ) && (
  <div style={{ backgroundColor: colorValue ? colorValue : value }} />
)}
```

File: src/components/ui/Alert/Alert.stories.mdx [hunk 5]
```diff
@@ -81,157 +69,98 @@ The `Alert` component uses [FastStore UI Alert](https://www.faststore.dev/refere
...
+        link={{ text: 'Really long link to add here', to: '/' }}
+        dismissible
+      >
+        Alert with a veeeery long description lorem ipsum dolor sit amet,
+        consectetur adipiscing elit. Donec sollicitudin erat non felis feugiat
+        porta. Nam vitae turpis metus. Maecenas feugiat justo tristique libero
+        ultricies.
+      </Alert>
+    }
+    discouragedDescription="Don't use really long texts. Some important information can be lost."
```
Comment (Mariana-Caetano) [inline src/components/ui/Alert/Alert.stories.mdx:164]:
```suggestion
    discouragedDescription="Don't use long texts. Important information can be lost."
```

Comment (Mariana-Caetano) [reply src/components/ui/Alert/Alert.stories.mdx:164]: What about defining a character limit for the text? Maybe something around 30 - 40 characters. So the person knows that something over that limit is considered a long text.

File: src/components/ui/Alert/Alert.stories.mdx [hunk 6]
```diff
@@ -81,157 +69,98 @@ The `Alert` component uses [FastStore UI Alert](https://www.faststore.dev/refere
...
+    recommendedUsage={
+      <Alert
+        icon={<Icon name="ShieldCheck" width={24} height={24} />}
+        link={{ text: 'Buy Now', to: '/' }}
+        dismissible
+      >
+        Free shipping starts at $300
+      </Alert>
+    }
+    recommendedDescription="Prefer short messages to inform about a behavior of a system, feature or page."
```
Comment (Mariana-Caetano) [inline src/components/ui/Alert/Alert.stories.mdx:151]:
```suggestion
    recommendedDescription="Write short messages to inform about the behavior of a system, feature, or page."
```

**Review by Mariana-Caetano — CHANGES_REQUESTED** [review]
Awesome, Renatinha! 🚀
I left some suggestions and comments.

**Review by hellofanny — APPROVED** [review]

File: .storybook/components/TokenRow.tsx (hunk 2 above)
Comment (lucasfp13) [reply .storybook/components/TokenRow.tsx]: yeah, both `value` and `colorValue` as optional props

**Review by lucasfp13 — APPROVED** [review]
Looks great, nice doc improvement ⭐

**Review by Mariana-Caetano — APPROVED** [review]

**Pull request merged**